From 554bdffe23cfd81a2f646b1460b4f643cfa1cdcd Mon Sep 17 00:00:00 2001
From: Franck LENORMAND <franck.lenormand@nxp.com>
Date: Thu, 19 Jul 2018 11:35:35 +0200
Subject: [PATCH 4211/5242] MLK-18082: ARM: dts: imx: Change size of caam-sm
 to correct size

commit  bd8125532f3bb61d4b14e799dddb04403efe432c from
https://source.codeaurora.org/external/imx/linux-imx.git

The resources retrieved by CAAM driver was wrong as the size
was not correct hence future uses might have issues.

before:
[    3.010744] caam 30900000.caam:
sm res: [start: 0000000000100000,
	end: 0000000000107ffe,
	name: /caam-sm@00100000,
	flags:0x200 desc:0x0] -> size: 0x7fff

modif to actual size:
[    3.012495] caam 30900000.caam:
sm res: [start: 0000000000100000,
	end: 0000000000107fff,
	name: /caam-sm@00100000,
	flags:0x200 desc:0x0] -> size: 0x8000

Signed-off-by: Franck LENORMAND <franck.lenormand@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/crypto/fsl-sec4.txt        |    2 +-
 arch/arm/boot/dts/imx6qdl.dtsi                     |    2 +-
 arch/arm/boot/dts/imx6sx.dtsi                      |    2 +-
 arch/arm/boot/dts/imx6ul.dtsi                      |    2 +-
 arch/arm/boot/dts/imx7d.dtsi                       |    2 +-
 arch/arm/boot/dts/imx7ulp.dtsi                     |    2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi      |    2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi      |    2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi      |    2 +-
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi      |    2 +-
 10 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/Documentation/devicetree/bindings/crypto/fsl-sec4.txt b/Documentation/devicetree/bindings/crypto/fsl-sec4.txt
index 7747810..a2ab1e0 100644
--- a/Documentation/devicetree/bindings/crypto/fsl-sec4.txt
+++ b/Documentation/devicetree/bindings/crypto/fsl-sec4.txt
@@ -278,7 +278,7 @@ Secure memory (SM) Node
 EXAMPLE
   caam_sm: caam-sm@00100000 {
     compatible = "fsl,imx6q-caam-sm";
-    reg = <0x00100000 0x3fff>;
+    reg = <0x00100000 0x4000>;
   };
 
 =====================================================================
diff --git a/arch/arm/boot/dts/imx6qdl.dtsi b/arch/arm/boot/dts/imx6qdl.dtsi
index e75b3e5..66ead9c 100644
--- a/arch/arm/boot/dts/imx6qdl.dtsi
+++ b/arch/arm/boot/dts/imx6qdl.dtsi
@@ -148,7 +148,7 @@
 
 		caam_sm: caam-sm@00100000 {
 			compatible = "fsl,imx6q-caam-sm";
-			reg = <0x00100000 0x3fff>;
+			reg = <0x00100000 0x4000>;
 		};
 
 		dma_apbh: dma-apbh@110000 {
diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 51e564d..62abf23 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -277,7 +277,7 @@
 
 		caam_sm: caam-sm@100000 {
 			compatible = "fsl,imx6q-caam-sm";
-			reg = <0x00100000 0x7fff>;
+			reg = <0x00100000 0x8000>;
 		};
 
 		irq_sec_vio: caam_secvio {
diff --git a/arch/arm/boot/dts/imx6ul.dtsi b/arch/arm/boot/dts/imx6ul.dtsi
index 3a29415..27e7756 100644
--- a/arch/arm/boot/dts/imx6ul.dtsi
+++ b/arch/arm/boot/dts/imx6ul.dtsi
@@ -220,7 +220,7 @@
 
 		caam_sm: caam-sm@00100000 {
 			compatible = "fsl,imx7d-caam-sm", "fsl,imx6q-caam-sm";
-			reg = <0x00100000 0x7fff>;
+			reg = <0x00100000 0x8000>;
 		};
 
 		irq_sec_vio: caam_secvio {
diff --git a/arch/arm/boot/dts/imx7d.dtsi b/arch/arm/boot/dts/imx7d.dtsi
index cd29cfc..0845fea 100644
--- a/arch/arm/boot/dts/imx7d.dtsi
+++ b/arch/arm/boot/dts/imx7d.dtsi
@@ -108,7 +108,7 @@
 
 		caam_sm: caam-sm@00100000 {
 			 compatible = "fsl,imx7d-caam-sm", "fsl,imx6q-caam-sm";
-			 reg = <0x00100000 0x7fff>;
+			 reg = <0x00100000 0x8000>;
 		};
 
 		irq_sec_vio: caam_secvio {
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index c4f5d9c..3f341bf 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -148,7 +148,7 @@
 
 	caam_sm: caam-sm@26000000 {
 			 compatible = "fsl,imx6q-caam-sm";
-			 reg = <0x26000000 0x7fff>;
+			 reg = <0x26000000 0x8000>;
 	};
 
 	ahbbridge0: ahb-bridge0@40000000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index 9aca176..12c67c2 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -3340,7 +3340,7 @@
 
 	caam_sm: caam-sm@31800000 {
 		compatible = "fsl,imx6q-caam-sm";
-		reg = <0 0x31800000 0 0x1ffff>;
+		reg = <0 0x31800000 0 0x10000>;
 	};
 
 	sc_pwrkey: sc-powerkey {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index 4b48a46..32e6a63 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -1186,7 +1186,7 @@
 
 	caam_sm: caam-sm@00100000 {
 		compatible = "fsl,imx6q-caam-sm";
-		reg = <0 0x00100000 0 0x7fff>;
+		reg = <0 0x00100000 0 0x8000>;
 	};
 
 	caam_snvs: caam-snvs@30370000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 37c2c48..6571151 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -1391,7 +1391,7 @@
 
 	caam_sm: caam-sm@00100000 {
 		compatible = "fsl,imx6q-caam-sm";
-		reg = <0 0x00100000 0 0x7fff>;
+		reg = <0 0x00100000 0 0x8000>;
 	};
 
 	caam_snvs: caam-snvs@30370000 {
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index a4fe441..be75f8c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -4179,7 +4179,7 @@
 
 	caam_sm: caam-sm@31800000 {
 		compatible = "fsl,imx6q-caam-sm";
-		reg = <0 0x31800000 0 0x1ffff>;
+		reg = <0 0x31800000 0 0x10000>;
 	};
 
 	sc_pwrkey: sc-powerkey {
-- 
1.7.9.5

